src/c-writer.cc

```diff
@@ -1385,21 +1390,35 @@ void CWriter::WriteGlobalInitializers() {
   Write(CloseBrace(), Newline());
 }
 
-void CWriter::WriteDataInitializers() {
-  const Memory* memory = nullptr;
-  Index data_segment_index = 0;
+static inline bool is_droppable(const DataSegment* data_segment) {
+  return (data_segment->kind == SegmentKind::Passive) &&
+         (!data_segment->data.empty());
+}
+
+static inline bool is_droppable(const ElemSegment* elem_segment) {
+  return (elem_segment->kind == SegmentKind::Passive) &&
+         (!elem_segment->elem_exprs.empty());
+}
+
+void CWriter::WriteDataInstances() {
+  for (const DataSegment* data_segment : module_->data_segments) {
+    DefineGlobalScopeName(data_segment->name);
+    if (is_droppable(data_segment)) {
+      Write("bool ", "data_segment_dropped_", GetGlobalName(data_segment->name),
+            " : 1;", Newline());
+    }
+  }
```

Comment:
I wonder if we should have some more examples of wasm2c output actual checked in so that when reviewing these changes we can see the effect they have on the output?  Not sure what the best way to do that is.

Reply:
I also like this idea but not sure how far to take it... some options are probably:
- check in rot13.c like we do fac.c?
- check in some representative outputs of running wasm2c on the spec testsuite?
- check in every single output of running wasm2c on the spec testsuite?

Reply:
How about a few hand-crafted tests under `test/wasm2c` that include the input wat and the output c code in the same same file.

Reply:
Did a first stab at this in #2006.